cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/IntraBrokerDiskUsageDistributionGoal.java

```diff
@@ -412,32 +403,22 @@ private void rebalanceBySwappingLoadOut(Disk disk,
 
     while (!candidateDiskPQ.isEmpty()) {
       Disk candidateDisk = candidateDiskPQ.poll();
-      candidateDisk.trackSortedReplicas(name(),
-                                        ReplicaSortFunctionFactory.selectOnlineReplicas(),
-                                        ReplicaSortFunctionFactory.sortByMetricGroupValue(RESOURCE.name()));
-      for (Iterator<Replica> iterator = disk.trackedSortedReplicas(name()).reverselySortedReplicas().iterator();
+      for (Iterator<Replica> iterator = disk.trackedSortedReplicas(replicaSortName(this, true, false)).sortedReplicas(false).iterator();
```

Comment:
Every time the swap succeed, we will [break out of the `for` loop](https://github.com/linkedin/cruise-control/blob/4c1979da27e5ad55ac462a385915bfedfac9e851/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/IntraBrokerDiskUsageDistributionGoal.java#L419). Next time we enter the `for` loop, we will have a consistent iterator, so the exception will not be thrown.

Reply:
Thanks for the clarification!
Can we add a unit test to ensure that any unintended future change to this behavior can be captured via tests rather than potentially during run time?

Reply:
Added the test.